cawg_identity/src/x509/x509_signature_verifier.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2025 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use async_trait::async_trait;
+use c2pa_crypto::{
+    cose::{CertificateInfo, CoseError, Verifier},
+    raw_signature::RawSignatureValidationError,
+};
+use c2pa_status_tracker::DetailedStatusTracker;
+
+use crate::{SignatureVerifier, SignerPayload, ValidationError};
+
+/// An implementation of [`SignatureVerifier`] that supports COSE signatures
+/// generated from X.509 credentials as specified in [§8.2, X.509 certificates
+/// and COSE signatures].
+///
+/// [`SignatureVerifier`]: crate::SignatureVerifier
+/// [§8.2, X.509 certificates and COSE signatures]: https://cawg.io/identity/1.1-draft/#_x_509_certificates_and_cose_signatures
+pub(crate) struct X509SignatureVerifier {}
+
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+impl SignatureVerifier for X509SignatureVerifier {
+    type Error = CoseError;
+    type Output = CertificateInfo;
+
+    async fn check_signature(
+        &self,
+        signer_payload: &SignerPayload,
+        signature: &[u8],
+    ) -> Result<Self::Output, ValidationError<Self::Error>> {
+        if signer_payload.sig_type != "cawg.x509.cose" {
```

Comment:
Should we consider putting the signature types somewhere in constants/enums for easier checking?

Reply:
Done now.